crawl-ref/source/melee-attack.cc

```diff
@@ -687,7 +687,7 @@ static void _hydra_consider_devouring(monster &defender)
  */
 bool melee_attack::handle_phase_killed()
 {
-    if (attacker->is_player() && you.form == transformation::hydra
+    if (attacker->is_player() && ( you.species == SP_DUSK_WALKER || you.form == transformation::hydra)
         && defender->is_monster() // better safe than sorry
```

Comment:
This line is too long: put the three checks on three separate lines like this:
`if (attacker->is_player()`
`----&& (you.species == SP_DUSK_WALKER`
`--------|| you.form == transformation::hydra)`
(the dashes are fillers for spaces, because otherwise it just ignores whitespace)